library/manageiq_provider.py

```diff
@@ -234,12 +252,13 @@ def find_provider_by_name(self, provider_name):
         providers = self.client.collections.providers
         return next((p.id for p in providers if p.name == provider_name), None)
 
-    def generate_endpoint(self, role, hostname, port, authtype, token):
+    def generate_endpoint(self, role, authtype, hostname=None, port=None, token=None, userid=None, password=None):
         """ Returns one provider endpoint hash.
         """
         return {'endpoint': {'role': role, 'hostname': hostname,
                              'port': port},
-                'authentication': {'role': authtype, 'auth_key': token}}
+                'authentication': {'authtype': authtype, 'auth_key': token,
+                                   'userid': userid, 'password': password}}
```

Comment:
The 'authentication' dict will contain some Nones, right?  I suppose they become `null` in json.  Is it safe to send irrelevant keys with `null` to the API?

I'm not sure this function is all that helpful now.
Perhaps it should just take the whole authentication dict as one param?